test/global_ctor_dtor.ll

```diff
@@ -0,0 +1,60 @@
+; RUN: llvm-as %s -o %t.bc
+; RUN: llvm-spirv %t.bc -spirv-ext=+all -spirv-text -o - | FileCheck %s
+; RUN: llvm-spirv %t.bc -spirv-ext=+all -o %t.spv
+; RUN: llvm-spirv -r %t.spv
+
```

Comment:
Maybe also add a `spirv-val` step at the end?

Reply:
Done

Reply:
Apparently the validator found something:
```error: line 35: ID 6[%_GLOBAL__sub_I_test_cpp_ctor] has not been defined

  %12 = OpConstantComposite %_struct_10 %uint_65535 %_GLOBAL__sub_I_test_cpp_ctor %9```

Reply:
SPIR-V spec "Forward references (an operand <id> that appears before the Result <id> defining it) are allowed for:
Operands that are an OpFunction. This allows for recursion and early declaration of entry points."
Do this mean the use of function ID 6[%_GLOBAL__sub_I_test_cpp_ctor] here is correct?

Reply:
Interesting, this might be shortcoming in spirv-val's `spvOperandCanBeForwardDeclaredFunction` then; it doesn't seem to handle `OpConstantComposite` at all.

Reply:
The `TopologicalSort` class is initialized with types, constants and global variables. How do functions get there in the first place?

Reply:
llvm.global_ctors/llvm.global_dtors contains a list of functions, they get into ConstAndVarVec.

Reply:
After adding SpvOpConstantComposite to spvOperandCanBeForwardDeclaredFunction, there is another failure:
error: line 28: Invalid use of function type result id 5[%5].
  %_ptr_Function_5 = OpTypePointer Function %5
spirv-val can't handle function pointer and some tests in transcoding/SPV_INTEL_function_pointers/ also have this failure.

Reply:
Until function pointers are supported by spirv-val we cannot add a spirv-val step I suppose, so LGTM.